src/PrestaShopBundle/Translation/Provider/ExternalModuleLegacySystemProvider.php

```diff
@@ -165,21 +210,127 @@ public function getXliffCatalogue()
+        $databaseCatalogue = new MessageCatalogue($this->locale);
+
+        foreach ($this->getTranslationDomains() as $translationDomain) {
+            if (!($this->getDatabaseLoader() instanceof DatabaseTranslationLoader)) {
+                continue;
+            }
+            $domainCatalogue = $this->getDatabaseLoader()->load(null, $this->locale, $translationDomain, $theme);
+
+            if ($domainCatalogue instanceof MessageCatalogue) {
+                $databaseCatalogue->addCatalogue($domainCatalogue);
+            }
+        }
+
+        return $databaseCatalogue;
     }
 
     /**
-     * {@inheritdoc}
+     * @return string Path to app/Resources/translations/{locale}
      */
     public function getResourceDirectory()
     {
         return $this->getDefaultResourceDirectory();
     }
 
+    /**
+     * @return LoaderInterface
+     */
+    public function getDatabaseLoader()
+    {
+        return $this->databaseLoader;
+    }
+
+    /**
+     * Empties out the catalogue by removing translations but leaving keys
+     *
+     * @param MessageCatalogueInterface $messageCatalogue
+     *
+     * @return MessageCatalogueInterface Empty the catalogue
+     */
+    public function emptyCatalogue(MessageCatalogueInterface $messageCatalogue)
+    {
+        foreach ($messageCatalogue->all() as $domain => $messages) {
+            foreach (array_keys($messages) as $translationKey) {
+                $messageCatalogue->set($translationKey, '', $domain);
+            }
+        }
+
+        return $messageCatalogue;
+    }
+
+    /**
+     * @param array $paths a list of paths when we can look for translations
+     * @param string $locale the Symfony (not the PrestaShop one) locale
+     * @param string|null $pattern a regular expression
+     *
+     * @return MessageCatalogue
+     *
+     * @throws FileNotFoundException
+     */
+    public function getCatalogueFromPaths($paths, $locale, $pattern = null)
+    {
+        return (new TranslationFinder())->getCatalogueFromPaths($paths, $locale, $pattern);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFilters()
+    {
+        return ['#^' . preg_quote($this->domain) . '([A-Z]|$)#'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTranslationDomains()
+    {
+        return ['^' . preg_quote($this->domain) . '([A-Z]|$)'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIdentifier(): string
+    {
+        return 'external_legacy_module';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setModuleName($moduleName)
+    {
+        if (null === $this->moduleName || empty($this->moduleName)) {
+            UnsupportedModuleException::moduleNotProvided(self::getIdentifier());
+        }
+
+        $this->moduleName = $moduleName;
+
+        // ugly hack, I know
```

Comment:
already there too
https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Translation/Provider/ExternalModuleLegacySystemProvider.php#L111
I guess it's ugly because domain is guessed from moduleName and not set explicitely

Reply:
If we have nothing else, it's not ugly for me 😄  we do what we can, no more.